src/game_inventory.cpp

```diff
@@ -639,13 +642,180 @@ item_location game_menus::inv::steal( player &p, player &victim )
                          string_format( _( "%s's inventory is empty." ), victim.name.c_str() ) );
 }
 
+class weapon_inventory_preset: public inventory_selector_preset
+{
+    public:
+        weapon_inventory_preset( const player &p ) : p( p ) {
+            append_cell( [ this ]( const item_location & loc ) {
+                if( !loc->is_gun() ) {
+                    return std::string();
+                }
+
+                const int total_damage = loc->gun_damage( true );
+
+                if( loc->ammo_data() && loc->ammo_remaining() ) {
+                    const int basic_damage = loc->gun_damage( false );
+                    const int ammo_damage = loc->ammo_data()->ammo->damage;
+
+                    return string_format( "%s<color_ltgray>+</color>%s <color_ltgray>=</color> %s",
+                                          get_damage_string( basic_damage, true ).c_str(),
+                                          get_damage_string( ammo_damage, true ).c_str(),
+                                          get_damage_string( total_damage, true ).c_str()
+                                        );
+                } else {
+                    return get_damage_string( total_damage );
+                }
+            }, _( "SHOT" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_damage_string( loc->damage_melee( DT_BASH ) );
+            }, _( "BASH" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_damage_string( loc->damage_melee( DT_CUT ) );
+            }, _( "CUT" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                return get_damage_string( loc->damage_melee( DT_STAB ) );
+            }, _( "STAB" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                if( deals_melee_damage( *loc ) ) {
+                    return good_bad_none( loc->type->m_to_hit );
+                }
+                return std::string();
+            }, _( "MELEE" ) );
+
+            append_cell( [ this ]( const item_location & loc ) {
+                if( deals_melee_damage( *loc ) ) {
+                    return string_format( "<color_yellow>%d</color>", loc->attack_time() );
```

Comment:
`attack_time` is incorrect: it's the base attack speed.
`g->u.attack_speed( *loc )` displays the correct time.